### PR TITLE
[WIP] HARVESTER: Harvester Cloud Provider compatibility check enhancement

### PR DESCRIPTION
**Pull request by n313893254**
### Summary
Harvester Cloud Provider compatibility check enhancement

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2753

### Screenshot/Video

image

### CODE REVIEW

**Comment by n313893254** [comment]
Hi, @mantis-toboggan-md. Backend PR does not merge yet. But I think you can review my PR, thanks.

File: shell/assets/translations/en-us.yaml [hunk 1]
```diff
@@ -1191,6 +1191,9 @@ cluster:
     warning:
       label: This is a Harvester Cluster - enable the Harvester feature flag to manage it
       state: Warning
+      cloudProvider: 
+        incompatible:
+          You cannot select Harvester cloud provider as the current Harvester Version is not compatible with RKE2 Kubernetes Version, click <a target="_blank" rel="noopener noreferrer nofollow" href="https://www.suse.com/suse-harvester/support-matrix/all-supported-versions">here</a> to see more detail.
```
Comment (mantis-toboggan-md) [inline shell/assets/translations/en-us.yaml:1196]: `You cannot select the Harvester cloud provider as the current Harvester version is not compatible with the selected RKE2 Kubernetes version, click <a target="_blank" rel="noopener noreferrer nofollow" href="https://www.suse.com/suse-harvester/support-matrix/all-supported-versions">here</a> to see which Kubernetes versions are supported.`

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 2]
```diff
@@ -1686,6 +1742,14 @@ export default {
         <Tab name="basic" label-key="cluster.tabs.basic" :weight="11" @active="refreshYamls">
           <Banner v-if="!haveArgInfo" color="warning" label="Configuration information is not available for the selected Kubernetes version.  The options available in this screen will be limited, you may want to use the YAML editor." />
           <Banner v-if="showk8s21LegacyWarning" color="warning" :label="t('cluster.legacyWarning')" />
+          <Banner
+            v-if="isHarvesterDriver && isHarvesterIncompatible"
```
Comment (mantis-toboggan-md) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:1746]: I think there should be `&& showCloudProvider` as well so the banner about cloud providers is not shown when the user cannot select a cloud provider (k3s kubernetes version is selected)

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]
I tested this by modifying setHarvesterVersionRange with the default value from [here](https://github.com/harvester/harvester/pull/2812):
```
 // const version = (res?.data || []).find(s => s.id === 'harvester-csi-ccm-version');

const version = { default: '{"harvester-cloud-provider": ">=0.0.1 <0.1.14", "harvester-csi-provider": ">=0.0.1 <0.1.15"}' };
 ```
 
 In light of code freeze being tomorrow, I'm fine with merging this without the backend merged yet.

**Review by mantis-toboggan-md — APPROVED** [review]

**Pull request merged**